geth/jail/jail.go

```diff
@@ -43,6 +48,85 @@ type Jail struct {
 	baseJSCode     string                     // JavaScript used to initialize all new cells with
 }
 
+// Copy returns a new JailCell instance with a new eventloop runtime associated with
+// the given cell.
+func (cell *JailCell) Copy() (common.JailCell, error) {
+	vmCopy := cell.vm.Copy()
+	loCopy := loop.New(vmCopy)
+
+	// Register fetch provider from ottoext.
+	if err := fetch.Define(vmCopy, loCopy); err != nil {
+		return nil, err
+	}
+
+	// Register event loop for timers.
+	if err := timers.Define(vmCopy, loCopy); err != nil {
+		return nil, err
+	}
+
+	return &JailCell{
+		id:  cell.id,
+		vm:  vmCopy,
+		lo:  loCopy,
+		sem: semaphore.New(1, JailCellRequestTimeout*time.Second),
+	}, nil
+}
+
+// Fetch attempts to call the underline Fetch API added through the
+// ottoext package.
+func (cell *JailCell) Fetch(url string, callback func(otto.Value)) (otto.Value, error) {
+	if err := cell.vm.Set("__captureFetch", func(res otto.Value) {
+		callback(res)
+	}); err != nil {
+		return otto.UndefinedValue(), err
+	}
+
+	return cell.Exec(`fetch("` + url + `").then(function(response){
+			__captureFetch({
+				"url": response.url,
+				"type": response.type,
+				"body": response.text(),
+				"status": response.status,
+				"headers": response.headers,
+			});
+		});
+	`)
+}
+
+// Exec evaluates the giving js string on the associated vm loop returning
+// an error.
+func (cell *JailCell) Exec(val string) (otto.Value, error) {
+	res, err := cell.vm.Run(val)
+	if err != nil {
+		return res, err
+	}
+
+	return res, cell.lo.Run()
+}
+
+// EvalExec evaluates the giving js string on the associated vm loop returning
+// an error.
+func (cell *JailCell) EvalExec(val string) error {
+	return cell.lo.EvalAndRun(val)
```

Comment:
Maybe it's better to proxy this call to `cell.Exec` and ignore the return value provided that `cell.lo.EvalAndRun` does exactly the same? This would add 1 line of code but improved code readability as the mentioned functionality is directly above.